jcl/src/java.base/share/classes/com/ibm/oti/util/Util.java

```diff
@@ -350,4 +382,44 @@ public static void printStackTraceElement(StackTraceElement e, Object elementSou
 	}
 }
 
+/**
+ * Helper method for printStackTraceElement to check if the classloader name
+ * should be included in the stack trace for the provided StackTraceElement.
+ *
+ * @param element The StackTraceElement to check
+ * @return true if the classloader name should be included, false otherwise
+ */
+private static boolean includeClassLoaderName(StackTraceElement element) {
+	return vmLangAccess.getIncludeClassLoaderName(element);
+}
+
+/**
+ * Helper method for printStackTraceElement to check if the module version
+ * should be included in the stack trace for the provided StackTraceElement.
+ *
+ * The module version should not be included for modules that are non-upgradeable.
+ *
+ * @param element The StackTraceElement to check
+ * @return true if the module version should be included, false otherwise
+ */
+private static boolean includeModuleVersion(StackTraceElement element) {
+	boolean includeModuleVersion = vmLangAccess.getIncludeModuleVersion(element);
+	boolean isNonUpgradeableModule = NonUpgradeableModules.moduleNames.contains(element.getModuleName());
+
+	return includeModuleVersion && !isNonUpgradeableModule;
+}
+private static class NonUpgradeableModules {
+	static Set<String> moduleNames;
+
+	static {
+		Optional<ResolvedModule> javaBaseModule = ModuleLayer.boot().configuration().findModule("java.base"); //$NON-NLS-1$
+
+		if (javaBaseModule.isPresent()) {
+			ResolvedModule resolvedJavaBaseModule = javaBaseModule.get();
+			ModuleReferenceImpl javaBaseModuleRef = (ModuleReferenceImpl) resolvedJavaBaseModule.reference();
+			moduleNames = javaBaseModuleRef.recordedHashes().names();
+		}
+	}
+}
```

Comment:
@pshipton These are the only new changes since the last review. This makes it so that we don't print the module version for non-upgradeable modules.